### PR TITLE
SYNR-1181

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: tests/testthat/test_table.R [hunk 1]
```diff
@@ -15,6 +15,17 @@ test_that("data.frame can be write to and read from csv consistently", {
   expect_equal(b, df2$b)
 })
 
+test_that("empty data.frame can be write to and read from csv consistently", {
```
Comment (kdaily) [inline tests/testthat/test_table.R]: 'can be written to'

Comment (kimyen) [reply tests/testthat/test_table.R]: thanks Kenny! I copied this so there is more to fix.

File: R/table.R [hunk 2]
```diff
@@ -5,18 +5,24 @@
...
+  if (is.data.frame(dataFrame) && nrow(dataFrame) != 0) {
+    for (i in 1:dim(dataFrame)[2]) {
+      if (is.numeric(dataFrame[[i]])) {
+        dataFrame[[i]][is.nan(dataFrame[[i]])]<-"NaN"
+      } else if (is(dataFrame[[i]], "POSIXct")) {
+        # convert POSIXct before uploading to Synapse
+        dataFrame[[i]]<-format(as.POSIXlt(dataFrame[[i]], 'UTC', usetz=TRUE), "%Y-%m-%d %H:%M:%S.000")
+      }
     }
+    write.csv(x=dataFrame, file=filePath, row.names=FALSE, na="")
```
Comment (brucehoff) [inline R/table.R]: I'm concerned that the behavior for an empty data frame is not as expected. The behavior is that saveToCsv does nothing while one would expect that it writes a file having no data rows.

File: R/table.R [hunk 3]
```diff
@@ -5,18 +5,24 @@
 
 # writing a dataFrame to a csv
 saveToCsv <- function(dataFrame, filePath) {
-  for (i in 1:dim(dataFrame)[2]) {
-    if (is.numeric(dataFrame[[i]])) {
-      dataFrame[[i]][is.nan(dataFrame[[i]])]<-"NaN"
-    } else if (is(dataFrame[[i]], "POSIXct")) {
-      # convert POSIXct before uploading to Synapse
-      dataFrame[[i]]<-format(as.POSIXlt(dataFrame[[i]], 'UTC', usetz=TRUE), "%Y-%m-%d %H:%M:%S.000")
+  if (is.data.frame(dataFrame) && nrow(dataFrame) != 0) {
```
Comment (brucehoff) [inline R/table.R]: if (!is.data.frame(dataFrame)) stop("dataFrame arg is not a data frame.")

Comment (brucehoff) [reply R/table.R]: if (nrow(dataFrame)==0) writeToCsv(**empty data frame**, filePath)

**Pull request merged**